api/routes/google_drive.py

```
@@ -0,0 +1,257 @@
+):
+    """
+    Google OAuth callback - exchange code for tokens and save to profile.
+    """
+    # Extract profile_id from state
+    try:
+        profile_id, _, _ = state.split(':', 2)
+    except ValueError:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Invalid state parameter",
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)